kv/util/util.go

```diff
@@ -223,10 +282,18 @@ func updateWithKV(getValue getValueFn, update updateFn, key string, v kv.Value,
 	newValue, err := getValue(v)
 	if err != nil {
 		update(defaultValue)
-		logInvalidUpdate(logger, key, v.Version(), defaultValue, err)
+		logMalformedUpdate(logger, key, v.Version(), defaultValue, err)
 		return err
 	}
 
+	if validate != nil {
```

Comment:
Feels a little weird that a malformed update applies the default value while a well-formed invalid value does not, maybe we should change the behavior to not apply default value in both cases?

Reply:
Sounds good to me, will update. I had left it this way since we applied the defaults for malformed previously, so I didn't want to change it until we talked about it.